src/scenes/CollectionCreationFlow/CollectionCreationFlow.tsx

```diff
@@ -1,12 +1,21 @@
 import { memo } from 'react';
-import { RouteComponentProps } from '@reach/router';
+import { Redirect, RouteComponentProps } from '@reach/router';
 import { Wizard, Steps, Step } from 'react-albus';
 import WizardFooter from './WizardFooter';
 import CreateFirstCollection from './steps/CreateFirstCollection';
 import AddNfts from './steps/AddNfts/AddNfts';
 import OrganizeCollections from './steps/OrganizeCollections';
+import { useAuthState } from 'contexts/auth/AuthContext';
+import { isLoggedInState } from 'contexts/auth/types';
 
 function CollectionCreationFlow(_: RouteComponentProps) {
+  const authState = useAuthState();
+  const isLoggedIn = isLoggedInState(authState);
+
+  if (!isLoggedIn) {
```

Comment:
maybe we want to put the redirect somewhere else so we don't have add it to every page

Reply:
Great call, I think reach router should have something around authenticated routing

Reply:
https://github.com/Robinnnnn/good-vibes-only-client/blob/master/src/scenes/Routes.tsx#L10

This is how I did it before, but I'm not sure if it's the right way

Reply:
It would be cool if we made a higher order component that's like PrivateRoute. Maybe we can pair on this next time